pkg/resource/transform_job/custom_delta.go

```diff
@@ -23,12 +23,16 @@ func customSetDefaults(
 	b *resource,
 ) {
 	// TransformInput is a required field.
-	if ackcompare.IsNil(a.ko.Spec.TransformInput.CompressionType) && ackcompare.IsNotNil(b.ko.Spec.TransformInput.CompressionType) {
-		a.ko.Spec.TransformInput.CompressionType = b.ko.Spec.TransformInput.CompressionType
+	if ackcompare.IsNotNil(a.ko.Spec.TransformInput) && ackcompare.IsNotNil(b.ko.Spec.TransformInput) {
```

Comment:
why this change? TransformInput is a required field and will never be nil

Reply:
Unit test was breaking due to nil pointer exception. We needed to check that the TransformInput.CompressionType and TransformInput.SplitType were not null before trying to access their fields. (Otherwise we got a nil pointer exception)

Reply:
I don't understand how. this code path is not executed for the test in this PR. there is only one test that is calling Create which is also invalid. Delta is only constructed before doing an update. 

Can you double-check this?